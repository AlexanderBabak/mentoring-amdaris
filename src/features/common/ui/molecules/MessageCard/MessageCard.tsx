import { FC, ReactNode } from "react";
import { Box, Typography } from "@mui/material";

export interface MessageCardProps {
  image: string;
  heading: string;
  message?: ReactNode;
  imgWidth?: string;
  imgHeight?: string;
}

const MessageCard: FC<MessageCardProps> = ({ image: Image, heading, message, imgWidth, imgHeight }) => {
  return (
    <Box
      data-testid="message-card"
      flex={1}
      display="flex"
      flexDirection="column"
      justifyContent="center"
      alignItems="center"
    >
      <img width={imgWidth || "132px"} height={imgHeight || "132px"} src={Image} alt={heading} />

      <Typography variant="h2" aria-label="Message heading" mt={2}>
        {heading}
      </Typography>

      {message}
    </Box>
  );
};

export default MessageCard;
