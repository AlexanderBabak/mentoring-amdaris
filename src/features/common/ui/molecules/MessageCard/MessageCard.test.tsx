import { render } from "@testing-library/react";
import placeholder from "features/common/assets/images/image-placeholder.svg";
import MessageCard from "./MessageCard";

describe("MessageCard", () => {
  it("should be rendered with the image", async () => {
    const { baseElement } = render(
      <MessageCard image={placeholder} heading="Start searching for user." message="Search for user by email." />,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
