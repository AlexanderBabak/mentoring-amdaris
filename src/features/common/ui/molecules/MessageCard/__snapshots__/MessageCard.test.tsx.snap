// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageCard should be rendered with the image 1`] = `
.emotion-0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 300;
  font-size: 3.75rem;
  line-height: 1.2;
  letter-spacing: -0.00833em;
  margin-top: 16px;
}

<body>
  <div>
    <div
      class="MuiBox-root emotion-0"
      data-testid="message-card"
    >
      <img
        alt="Start searching for user."
        height="132px"
        width="132px"
      />
      <h2
        aria-label="Message heading"
        class="MuiTypography-root MuiTypography-h2 emotion-1"
      >
        Start searching for user.
      </h2>
      Search for user by email.
    </div>
  </div>
</body>
`;
